PWGHF/TableProducer/treeCreatorD0ToKPi.cxx

```diff
@@ -242,110 +229,53 @@ struct HfTreeCreatorD0ToKPi {
-            candidate.errorDecayLengthXY(),
-            candidate.chi2PCA(),
-            candidate.rSecondaryVertex(),
-            candidate.decayLength(),
-            candidate.decayLengthXY(),
-            candidate.decayLengthNormalised(),
-            candidate.decayLengthXYNormalised(),
-            candidate.impactParameterNormalised0(),
-            candidate.ptProng0(),
-            RecoDecay::p(candidate.pxProng0(), candidate.pyProng0(), candidate.pzProng0()),
-            candidate.impactParameterNormalised1(),
-            candidate.ptProng1(),
-            RecoDecay::p(candidate.pxProng1(), candidate.pyProng1(), candidate.pzProng1()),
-            candidate.pxProng0(),
-            candidate.pyProng0(),
-            candidate.pzProng0(),
-            candidate.pxProng1(),
-            candidate.pyProng1(),
-            candidate.pzProng1(),
-            candidate.impactParameter0(),
-            candidate.impactParameter1(),
-            candidate.errorImpactParameter0(),
-            candidate.errorImpactParameter1(),
-            candidate.prong0_as<aod::BigTracksPID>().tpcNSigmaPi(),
-            candidate.prong0_as<aod::BigTracksPID>().tpcNSigmaKa(),
-            candidate.prong0_as<aod::BigTracksPID>().tofNSigmaPi(),
-            candidate.prong0_as<aod::BigTracksPID>().tofNSigmaKa(),
-            candidate.prong1_as<aod::BigTracksPID>().tpcNSigmaPi(),
-            candidate.prong1_as<aod::BigTracksPID>().tpcNSigmaKa(),
-            candidate.prong1_as<aod::BigTracksPID>().tofNSigmaPi(),
-            candidate.prong1_as<aod::BigTracksPID>().tofNSigmaKa(),
-            1 << CandFlag,
-            FunctionInvMass,
-            candidate.impactParameterProduct(),
-            FunctionCosThetaStar,
-            candidate.pt(),
-            candidate.p(),
-            candidate.cpa(),
-            candidate.cpaXY(),
-            FunctionCt,
-            candidate.eta(),
-            candidate.phi(),
-            FunctionY,
-            FunctionE,
-            candidate.flagMcMatchRec(),
-            candidate.originMcRec());
-        }
-      };
-
-      fillTable(0, candidate.isSelD0(), invMassD0ToPiK(candidate), cosThetaStarD0(candidate), ctD0(candidate), yD0(candidate), eD0(candidate));
-      fillTable(1, candidate.isSelD0bar(), invMassD0barToKPi(candidate), cosThetaStarD0bar(candidate), ctD0(candidate), yD0(candidate), eD0(candidate));
+      auto prong0 = candidate.prong0_as<aod::BigTracksPID>();
+      auto prong1 = candidate.prong0_as<aod::BigTracksPID>();
+      fillTable(candidate, prong0, prong1, 0, candidate.isSelD0(), invMassD0ToPiK(candidate), cosThetaStarD0(candidate), ctD0(candidate), yD0(candidate), eD0(candidate), candidate.flagMcMatchRec(), candidate.originMcRec());
+      fillTable(candidate, prong0, prong1, 1, candidate.isSelD0bar(), invMassD0barToKPi(candidate), cosThetaStarD0bar(candidate), ctD0(candidate), yD0(candidate), eD0(candidate), candidate.flagMcMatchRec(), candidate.originMcRec());
```

Comment:
The function `fillTable` does not need so many arguments. Providing `candidate, prong0, prong1, 0` is enough. The rest can be retrieved inside the function from `candidate`

Reply:
Actually, you can keep it this way and avoid calculating `ctD0` and `yD0` twice.

Reply:
Please address the comment and calculate `ctD0` and `yD0` only once.